cpp/src/arrow/python/pandas_convert.cc

```diff
@@ -1256,6 +1256,66 @@ inline Status ConvertFixedSizeBinary(const ChunkedArray& data, PyObject** out_va
   return Status::OK();
 }
 
+inline Status ConvertStruct(const ChunkedArray& data, PyObject** out_values) {
+  PyAcquireGIL lock;
+  OwnedRef dict_item;
+  std::vector<OwnedRef> fields_data;
+  if (data.num_chunks() <= 0) {
+    return Status::OK();
+  }
+  // ChunkedArray has at least one chunk
+  auto arr = static_cast<StructArray*>(data.chunk(0).get());
+  // Use it to cache the struct type and number of fields for all chunks
+  auto num_fields = arr->fields().size();
+  auto array_type = arr->type();
```

Comment:
`const` here provides a bit more information to the compiler, though in theory it should be able to deduce that these values are not changing during this call.